src/hooks/use-sync-states-progress-info.ts

```diff
@@ -130,12 +137,59 @@ export function useSyncStatesProgressInfo() {
 		[]
 	);
 
+	const getBackupStatusWithProgress = useCallback(
+		(
+			hasBackupCompleted: boolean,
+			pullStatesProgressInfo: PullStateProgressInfoValues,
+			response: SyncBackupResponse
+		) => {
+			const frontendStatus = hasBackupCompleted
+				? pullStatesProgressInfo.downloading.key
+				: response.status;
+			let newProgressInfo: PullStateProgressInfo | null = null;
+			if ( response.status === 'in-progress' ) {
+				newProgressInfo = pullStatesProgressInfo[ frontendStatus ];
+				newProgressInfo.progress =
+					IN_PROGRESS_INITIAL_VALUE + IN_PROGRESS_TO_DOWNLOADING_STEP * ( response.percent / 100 );
+			}
+			const statusWithProgress =
+				newProgressInfo ||
+				pullStatesProgressInfo[ frontendStatus ] ||
+				pullStatesProgressInfo.failed;
+
+			return statusWithProgress;
+		},
+		[]
+	);
+
+	const getPullStatusWithProgress = useCallback(
+		( sitePullState?: PullStateProgressInfo, importState?: ImportProgressState[ string ] ) => {
+			if ( ! importState && sitePullState ) {
+				return { message: sitePullState.message, progress: sitePullState.progress };
+			}
+			if ( importState ) {
+				if ( importState.progress === 100 ) {
+					return { message: __( 'Applying final details…' ), progress: 99 };
+				}
+				return {
+					message: importState.statusMessage,
+					progress:
+						IMPORTING_INITIAL_VALUE + IMPORTING_TO_FINISHED_STEP * ( importState.progress / 100 ),
```

Comment:
What do we accomplish with this calculation? `IMPORTING_INITIAL_VALUE` is always 80 and `IMPORTING_TO_FINISHED_STEP` is always 20, so it looks like we're saying `100 * (80 / 100)`. Why not just return `importState.progress`..?

Reply:
Not really, `80 + 20 * progress / 100 != progress` 😛 . This tries to move from 80% to 100% progressively determined by `importState.progress`

Reply:
Ofc, my bad 🤦‍♂️ Doesn't seem as bad now that I come back to it, though. Maybe assigning the `IMPORTING_TO_FINISHED_STEP * ( importState.progress / 100 )` part to a separate variable first would be enough to make it easier to understand.